client/app/components/packages/pas-form/edit.js

```diff
@@ -64,19 +68,29 @@ export default class PasFormComponent extends Component {
     return this.submittableChanges.isValid;
   }
 
-  @action
-  async save() {
-    await this.saveableChanges.save();
-
-    this.args.onSave();
+  get isSavingOrSubmitting() {
+    return this.save.isRunning || this.submit.isRunning;
   }
 
-  @action
-  async submit() {
-    await this.submittableChanges.save();
+  @task(function* () {
+    yield this.saveableChanges.execute();
 
-    this.args.onSubmit();
-  }
+    yield this.args.onSave();
+
+    yield this.saveableChanges.rollback();
+  })
+  save;
+
+  @task(function* () {
+    yield this.submittableChanges.execute();
+
+    yield this.args.onSubmit();
+
+    yield this.submittableChanges.rollback();
+
+    this.router.transitionTo('packages.show', this.package.id);
```

Comment:
I thought we refactored this out in another PR? Why go backwards on that progress?

Reply:
If it's because things aren't being awaited, you probably need to make those controller methods async.

Reply:
The reason I thought to be it here in the task is that it needs to happen _after_ the `rollback()` call. 

If it was within `this.args.onSubmit`, the transition would occur before `rollback()`

hmm, though now it just occurred to me that I may be able to rollback() immediately after execute()